talentmap_api/fsbid/services/assignment_cycles.py

```diff
@@ -317,3 +324,131 @@ def delete_assignment_cycle_req_mapping(request):
         "i_cycle_last_updt_user_id": timestamp_id
     }
     return mapped_request
+
+
+# --------------------------------------------------------------------------------------- Cycle Positions
+
+
+def get_cycle_positions_filters(jwt_token):
+    '''
+    Gets Filters for Cycle Positions Page
+    '''
+    args = {
+        'proc_name': 'qry_lstfsbidSearch',
+        'package_name': 'PKG_WEBAPI_WRAP',
+        'request_body': {},
+        'request_mapping_function': cycle_positions_filter_req_mapping,
+        'response_mapping_function': cycle_positions_filter_res_mapping,
+        'jwt_token': jwt_token,
+    }
+    return services.send_post_back_office(
+        **args
+    )
+
+
+def cycle_positions_filter_req_mapping(request):
+    return {
+        'PV_API_VERSION_I': '',
+        'PV_AD_ID_I': '',
+    }
+
+
+def cycle_positions_filter_res_mapping(data):
+    if data is None or (data['O_RETURN_CODE'] and data['O_RETURN_CODE'] is not 0):
+        logger.error(f"Fsbid call for Cycle Positions filters failed.")
+        return None
+
+    def bureau_map(x):
+        return {
+            'description': x.get('ORGS_SHORT_DESC'),
+        }
+
+    def org_map(x):
+        return {
+            'code': x.get('ORG_CODE'),
+            'description': f"{x.get('ORGS_SHORT_DESC')} ({x.get('ORG_CODE')})",
+        }
+
+    def skills_map(x):
+        return {
+            'code': x.get('SKL_CODE'),
+            'description': x.get('SKL_DESC'),
+        }
+
+    def grade_map(x):
+        return {
+            'code': x.get('GRD_CD'),
+            'description': x.get('GRD_DESCR_TXT'),
+        }
+
+    return {
+        'bureauFilters': list(map(bureau_map, data.get('QRY_LSTBUREAUS_DD_REF'))),
+        'orgFilters': list(map(org_map, data.get('QRY_LSTORGSHORT_DD_REF'))),
+        'skillsFilters': list(map(skills_map, data.get('QRY_LSTSKILLCODES_DD_REF'))),
+        'gradeFilters': list(map(grade_map, data.get('QRY_LSTGRADES_DD_REF'))),
+    }
+
+
+def get_cycle_positions(jwt, req):
+    '''
+    Gets Positions for the Cycle Positions Page
+    '''
+    args = {
+        'proc_name': 'qry_modCyclePos',
+        'package_name': 'PKG_WEBAPI_WRAP_SPRINT98',
+        'request_body': req,
+        'request_mapping_function': cycle_positions_req_mapping,
+        'response_mapping_function': cycle_positions_res_mapping,
+        'jwt_token': jwt,
+    }
+    return services.send_post_back_office(
+        **args
+    )
+
+
+def cycle_positions_req_mapping(req):
+    value = req.get('query')
+    grades = ",".join([item['code'] for item in value['selectedGrades']]) or ''
```

Comment:
once we sanitize on the FE - likely `req` but not sure
```suggestion
    grades = req.get('selectedGrades') or ''
```

Reply:
just kept as a `post` 🤷

Reply:
scratch that - totally get what you were saying about bug fixing & using the same patterns - updated to get + stringified the request from the FE